### ISSUE TITLE
Anomaly score for anomaly detection

### ISSUE THREAD

**Issue by steveazzolin** [aapatel09/handson-unsupervised-learning#11]
I have to configure an anomaly detection system. I found very useful what you wrote but I have a doubt.
I train PCA on the the train set. Then I get anomalyScores with the following function

```
def anomalyScores(originalDF, reducedDF):
    loss = np.sum((np.array(originalDF)-np.array(reducedDF))**2, axis=1)
    loss = pd.Series(data=loss,index=originalDF.index)
    loss = (loss-np.min(loss)) / (np.max(loss)-np.min(loss))
    return loss
``` 

At this point I try to use this trained algorithm for the test set.

The point is that I'am not sure that  `anomalyScores` so defined is suitble for this task.
Maybe using a MinMaxScaler would be better (or using this function but annotating the minimum and maximum obtained during the first evaluation) ?
This because consider the following example at inference time:

``` 
#during training I got the following losses
rec_error = np.array([0, 0.1, 1])
loss = (loss-np.min(loss)) / (np.max(loss)-np.min(loss)) #returns [0, 0.1, 1]

#now during testing suppose I got these reconstruction errors
rec_error = np.array([0, 0.1, 0.2])
loss = (loss-np.min(loss)) / (np.max(loss)-np.min(loss)) #returns [0, 0.5, 1]
``` 

So a reconstruction error of 0.1 wuold be considered much larger in the second example even if the trained model is the same.
While with a MinMaxScaler I would get:

``` 
from sklearn.preprocessing import MinMaxScaler
from sklearn.utils.validation import check_is_fitted

rec_error_train = np.array([ [0.0], [0.1], [1]])
rec_error_test = np.array([ [0.0], [0.1], [0.2]])
scaler = MinMaxScaler((0,1)).fit(rec_error_train)

loss_train = scaler.transform(rec_error_train) #returns [0, 0.1, 1]
loss_test = scaler.transform(rec_error_test) #returns [0, 0.1, 0.2]
``` 

What do you think?

**Comment by aapatel09** [comment]
I agree with this approach. Will try to incorporate this into the next update. Thanks.
